src/parquet/thrift/CMakeLists.txt

```diff
@@ -22,13 +22,29 @@ set(THRIFT_SRCS
 set_source_files_properties(parquet_types.cpp PROPERTIES
   COMPILE_FLAGS -Wno-unused-variable)
 
-add_library(parquet_thrift STATIC
-  ${THRIFT_SRCS}
-)
+if (PARQUET_BUILD_SHARED)
+  add_library(parquet_thrift_shared SHARED
+    ${THRIFT_SRCS}
+  )
```

Comment:
If the problem isn't reproducible starting from a new build of Thrift with `-fPIC`, setting `THRIFT_HOME` (clearing out any old Thrift installations before build/install), and a clean build, then I'm not sure how to proceed. Here is a build recipe that I know to produce the right `libthrift.a` -- I have not tested it with Thrift 0.10 yet: https://github.com/conda-forge/thrift-cpp-feedstock/blob/master/recipe/build.sh. I just created https://issues.apache.org/jira/browse/PARQUET-849 to upgrade the Thrift in our default toolchain